mcstatus/status_response.py

```diff
@@ -0,0 +1,662 @@
+
+    class Version(JavaStatusVersion):
+        """Deprecated class for `version` field.
+
+        Use `JavaStatusVersion` instead.
+        """
+
+        @overload
+        def __init__(self, raw: Dict[str, Any]) -> None:
+            ...  # pragma: no cover
+
+        @overload
+        def __init__(self, name: str, protocol: int) -> None:
+            ...  # pragma: no cover
+
+        def __init__(self, *args, **kwargs) -> None:
+            try:
+                # have the same signature
+                bound = _OLD_JAVA_INIT_SIGNATURE.bind(*args, **kwargs)
+
+                deprecated(
+                    lambda: None,
+                    replacement="mcstatus.status_response.JavaStatusVersion",
+                    date="2022-08",
+                    display_name="JavaStatusResponse.Version",
+                )()
+
+                # build and copy instance from `build` method
+                super().__init__(**self.build(bound.arguments["raw"]).__dict__)
+            except TypeError:
+                super().__init__(*args, **kwargs)
+
+        def __eq__(self, other):
+            return custom_eq(self, other)
+
+        def __repr__(self):
+            return super().__repr__().replace("JavaStatusResponse.Version", "JavaStatusVersion")
+
+    players: Players
+    version: Version
+
+    @overload
+    def __init__(self, raw: Dict[str, Any]) -> None:
+        ...  # pragma: no cover
+
+    @overload
+    def __init__(
+        self,
+        players: JavaStatusPlayers,
+        version: JavaStatusVersion,
+        motd: str,
+        latency: float,
+        icon: Optional[str],
+    ) -> None:
+        ...  # pragma: no cover
+
+    def __init__(self, *args, **kwargs) -> None:
+        try:
+            bound = _OLD_JAVA_INIT_SIGNATURE.bind(*args, **kwargs)
+
+            deprecated(lambda: None, replacement="build", date="2022-08", display_name="JavaStatusResponse.__init__")()
+
+            instance = self.build(bound.arguments["raw"]).__dict__
+            instance["players"] = self.Players(
+                online=instance["players"].online,
+                max=instance["players"].max,
+                sample=instance["players"].sample,
+            )
+            instance["version"] = self.Version(
+                name=instance["version"].name,
+                protocol=instance["version"].protocol,
+            )
+            super().__init__(**instance)
+        except TypeError:
+            super().__init__(*args, **kwargs)
+            # re-def fields to support the old interface
+            self.players = self.Players(
+                online=self.players.online,
+                max=self.players.max,
+                sample=self.players.sample,
+            )
+            self.version = self.Version(
+                name=self.version.name,
+                protocol=self.version.protocol,
+            )
+
+    @property
+    @deprecated(replacement="motd", date="2022-08")
```

Comment:
However, we creating API for **users**, not for protocol nitpickers. In Minecraft comunity no one uses `description` to name `motd`, everyone just uses simple *message of the day*. This also helps to follow OOP and inheritance.

No reason to name it different from that the community chose.

Reply:
This PR tries to standartise answer objects, this is the reason why I renamed it to `motd`. We should use inheriense instead of useless aliases.

Reply:
Even after re-read of all open discussions and found that I really too often just repeat myself, and quite ignore your arguments, I still do not see a reason to add useless aliases. Inheritance should be more desirable than following protocol's naming.

> Referencing the website of a non-vanilla implementation of Minecraft does not change my mind. 

We support non-vanilla implementations of Minecraft servers as well as vanilla. Because no one uses vanilla servers for production because it doesn't have plugins and very bad in optimization if compare with non-vanilla ones. You can look at https://bstats.org/global/bukkit (built-in metrics plugin in Bukkit (and its forks) site), there are, 161021 running servers on non-vanilla cores at the moment.

Even many of our bug reports (and bugs) are related to non-vanilla servers.

Reply:
From my scanning, I found about 500k servers hosted on the open internet on port 25565, only about 150k of them were non-vanilla, still leaving 350k servers hosted with the official vanilla server implementation (at least according to the version sent from status, which I'll admit can be modified and isn't completely trust-worthy, also this number includes forge/modded servers). 

Either way, mcstatus primarily supports mincraft's protocol, as it is a wrapper of it. The non-default implementations just happen to use the same protocol, because they obviously need the same clients to be able to connect. But if there is something non-standard in these servers, it's no surprise that mcstatus would fail, or somehow not conform to this.

Because mcstatus is a wrapper library around minceraft's protocol, we should adhere to it's naming and to the way it's done in general, not to the way it's done in some non-standard cases. It's true that most non-standard implementations call the description field in the JSON data from protocol status response packet `motd`, and players also commonly know it as such, but that's just why it can be an alias to description, but there's no real justification in removing `description` just because there's a lot of non-vanilla servers that call it differently.

Reply:
Added `description` in bb6289a411dc1ee7ab3b6185eb2e9cd20b5b6a70.